File: front/src/app/features/articles/interfaces/article.ts
export interface Article {
  id: number;
  title: string;
  content: string;
  username: string;
  userId: number;
  themeTitle: string;
  themeId: number;
  commentIds: number[];
  created_at: Date;
  updated_at: Date;
}

export interface Comment {
  id: number;
  content: string;
  userId: number;
  username: string;
  articleId: number;
  created_at: Date;
  updated_at: Date;
}

File: back/src/main/java/com/openclassrooms/mddapi/models/Article.java
package com.openclassrooms.mddapi.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "ARTICLES")
@Data
@Accessors(chain = true)
@EntityListeners(AuditingEntityListener.class)
@AllArgsConstructor
@NoArgsConstructor
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String content;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "theme_id", nullable = false)
    private Long themeId;

    @ElementCollection(fetch = FetchType.EAGER)
    private List<Long> commentIds = new ArrayList<>(); //Comment

    @CreationTimestamp
    private LocalDateTime created_at;

    @UpdateTimestamp
    private LocalDateTime updated_at;
}
